std/stdio.d

```diff
@@ -2880,10 +2880,9 @@ $(D Range) that locks the file and allows fast writing to it.
             }
 
             // put each element in turn.
-            alias Elem = Unqual!(ElementType!A);
-            foreach (Elem c; writeme)
+            for (; !writeme.empty; writeme.popFront)
             {
-                put(c);
+                put(writeme.front);
```

Comment:
Are you sure this actually makes a difference in the generated asm?  I'd like to see the asm output before approving this.

Using raw for-loops instead of foreach IMO is a step backward, and should not be done unless there's a very good reason for it.  My impression is that a properly-optimizing compiler would eliminate the supposedly-redundant loads easily. (And if not, it's the *compiler* that should be fixed; the user code should not be uglified to work around a compiler deficiency.) Don't be so quick to believe intermediate pseudo-source code representations of the compiled code; always check the asm to see what's *actually* being executed at runtime.

Also, I'd like to see what gdc/ldc outputs for both the foreach loop and the raw for loop. Dmd's optimizer isn't the best when it comes to loop optimizations.